server/controllers/postController.ts

```diff
@@ -5,20 +5,33 @@ const { postManager, userManager } = initManagers();
 async function postPosts(req, res, next) {
 	const refreshToken = req.cookies.refreshToken;
 	const username = await userManager.findRefreshToken(refreshToken);
-	const result = await postManager.addPost(req.body.title, req.body.body, username);
+	let result;
+	if (req.body.parent) {
+		result = await postManager.addPost(req.body.title, req.body.body, username, req.body.parent);
+	} else {
+		result = await postManager.addPost(req.body.title, req.body.body, username);
+	}
 	res.statusCode = 200;
 	res.send(result);
 };
 
 async function getPosts(req, res, next) {
+	let result;
 	if ('number' in req.query && 'page' in req.query) {
-		const { query: { number }, query: { page }, query: { sort } } = req;
-		const result = await postManager.getPostsPage(number, page, sort);
-		res.send(result);
+		const { query: { number }, query: { page }, query: { sort }, query: { search } } = req;
+		const filterObject: any = {};
+		if (sort !== undefined) filterObject.sort = sort;
+		if (search !== undefined) filterObject.search = search;
```

Comment:
Unfortunately, it is not obvious what are the types here. I assume just strings, so we don't need an explicit undefined check.

```suggestion
		if (sort) filterObject.sort = sort;
		if (search) filterObject.search = search;
```